velox/functions/sparksql/Hash.cpp

```diff
@@ -26,21 +26,270 @@ namespace {
 
 const int32_t kDefaultSeed = 42;
 
+// Computes the hash value of input using the hash function in HashClass.
+template <typename HashClass, typename SeedType, typename ReturnType>
```

Comment:
I think my point was that if you add the traits to the HashClass itself, can't you here just do something like:

```
template <typename THash>
THash::ReturnType hashOne(int32_t input, THash::SeedType seed) {
  return THash::hashInt32(input, seed);
}
```